app/src/utils/DatasetTypes.ts

```diff
@@ -0,0 +1,47 @@
+/* TODO: These are not yet comprehensive, final interfaces */
+
+export interface DatasetMetadata {
+  id: string;
+  name: string;
+  description: string;
+  fields: Field[];
+}
+
+export interface Field {
+  data_type: string;
+  name: string;
+  description: string;
+  origin_dataset: string;
+}
+
+// TODO: make typedef for valid data types instead of any.
+export type Row = Record<string, any>;
+
+export class Dataset {
+  readonly rows: Row[];
+  readonly metadata: DatasetMetadata;
+
+  constructor(rows: Row[], metadata: DatasetMetadata) {
+    this.rows = rows;
+    this.metadata = metadata;
+  }
+
+  getTableViewColumns() {
+    return this.metadata.fields.map((field: Field) => ({
+      Header: field.name,
+      accessor: field.name,
+    }));
+  }
+}
+
+export type AllMetadata = Record<string, DatasetMetadata>;
+
+export type LoadStatus = "unloaded" | "loading" | "loaded" | "error";
```

Comment:
Could this be an enum? https://www.typescriptlang.org/docs/handbook/enums.html
I think it's easier to make mistakes when relying on strings to match whereas with enums you'll have compiler problems if you don't use a valid value

Reply:
As far as typechecking goes, enums and type unions provide mostly equivalent compile-time safety. If you try to assign an arbitrary string to a variable of type `"unloaded" | "loading" | "loaded" | "error"` you'll get a compiler error. I did a bit of research on this and it seems like this is mostly a matter of preference. The main reason to use an enum would be if you need to iterate over the possible values, and if you prefer it for readability.

I have a preference for a type union for simple cases like this because you don't need to import the type every time to use it and because it doesn't generate weird transpiled JS which could make debugging difficult.

LMK if you have concerns with that.

https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant
https://medium.com/@maxheiber/alternatives-to-typescript-enums-50e4c16600b1